### ISSUE TITLE
🚀 [Feature] Username validation rules

### ISSUE THREAD

**Issue by ulfgebhardt** [gradido/gradido#1069]
## 🚀 Feature

The only requirement in code for an username is a minimum length of 2
```
    const MIN_CHARACTERS_USERNAME = 2
    if (username.length < MIN_CHARACTERS_USERNAME) { ... }
```
This is hardly sufficient. Especially when choosing two spaces for a username the problem becomes obvious.

**Comment by clauspeterhuebner** [comment]
Decition in Meeting 28.12.2021 16:00:
========================

Definition Pattern for new "username":
"communityname/useralias"
  * communityname
    - URL allowed signs
      * A-Z, a-z, 0-9
      * '-', '.', '_', '~'
  * useralias:
    - min 5 signs 
      * alphanumerisch
      * keine Umlaute
      * '-' und '_' sind nur zwischen sonst gültigen Zeichen erlaubt ([a-zA-Z0-9]-|_[a-zA-Z0-9])
    - blacklist of keywords, which have to be defined
    - reserved system relevant names must have max 4 signs

**Comment by clauspeterhuebner** [comment]
dependency to #1069 only set to hightlight the same topic in both tickets

**Comment by clauspeterhuebner** [comment]
will be solved with alias-concept, see also #2849
ticket will be closed
